base/uk.ac.stfc.isis.ibex.ui.journalviewer/src/uk/ac/stfc/isis/ibex/ui/journalviewer/JournalViewerView.java

```diff
@@ -140,80 +196,144 @@ protected void addColumns() {
+        table.setSize(prevSize.x, prevSize.y - 1);
+        table.setSize(prevSize);
+        table.setRedraw(true);
+    }
+    
+    private void setProgressIndicatorsVisible(final boolean visible) {
+        DISPLAY.asyncExec(() -> progressBar.setVisible(visible));
+    }
+    
+    private void resetPageNumber() {
+        DISPLAY.asyncExec(() -> spinnerPageNumber.setSelection(1));
+    }
+
     private void bind() {
         bindingContext.bindValue(WidgetProperties.text().observe(lblError),
                 BeanProperties.value("message").observe(model));
         bindingContext.bindValue(WidgetProperties.text().observe(lblLastUpdate),
                 BeanProperties.value("lastUpdate").observe(model));
-        bindingContext.bindValue(WidgetProperties.selection().observe(spinnerPageNumber),
-                BeanProperties.value("pageNumber").observe(model));
         bindingContext.bindValue(WidgetProperties.maximum().observe(spinnerPageNumber),
                 BeanProperties.value("pageNumberMax").observe(model));
 
+        spinnerPageNumber.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                setProgressIndicatorsVisible(true);
+                model.setPageNumber(spinnerPageNumber.getSelection()).thenAccept(ignored -> setProgressIndicatorsVisible(false));
+            }
+        });
         
         btnRefresh.addSelectionListener(new SelectionAdapter() {
             @Override
             public void widgetSelected(SelectionEvent e) {
-                model.refresh();
+                resetPageNumber();
+                setProgressIndicatorsVisible(true);
+                model.setPageNumber(1).thenAccept(ignored -> setProgressIndicatorsVisible(false));
             }
         });
         
-        model.addPropertyChangeListener("runs", new PropertyChangeListener() {
-			@Override
-			public void propertyChange(PropertyChangeEvent evt) {		
-				Display.getDefault().asyncExec(new Runnable() {
-					@Override
-					public void run() {
-						changeTableColumns();	
-						table.setRows(model.getRuns());
-					}
-				});
-			}
-		});
+        btnSearch.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                resetPageNumber();
+                search();
+            }
+        });
         
+        btnClear.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                resetPageNumber();
+                searchInput.clearInput();
+                model.resetActiveParameters();
+                setProgressIndicatorsVisible(true);
+                model.setPageNumber(1).thenAccept(ignored -> setProgressIndicatorsVisible(false));
+            }
+        });
+
+        model.addPropertyChangeListener("runs", new PropertyChangeListener() {
```

Comment:
You can get rid of some boiler plate here:
```
        model.addPropertyChangeListener("runs", e -> 
                DISPLAY.asyncExec(() -> {
                        setProgressIndicatorsVisible(true);
                        changeTableColumns();
                        table.setRows(model.getRuns());
                        setProgressIndicatorsVisible(false);
                    }));
```

Reply:
Done